modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/processor/PortletRegistryImpl.java

```diff
@@ -127,12 +127,30 @@ public List<String> getFragmentEntryLinkPortletIds(
 
 	@Override
 	public List<String> getPortletAliases() {
-		return new ArrayList<>(_portletNames.keySet());
+		return new ArrayList<>(_aliasPortletNames.keySet());
 	}
 
 	@Override
 	public String getPortletName(String alias) {
-		return _portletNames.get(alias);
+		return _aliasPortletNames.get(alias);
+	}
+
+	@Override
+	public void registerAlias(Map<String, Object> properties) {
+		_aliasPortletNames.put(
```

Comment:
@W2jx Please pass in the value directly rather than a map to avoid unnecessary retrieving.